src/internal/ParticleProcessor.ts

```diff
@@ -191,6 +201,24 @@ export class ParticleProcessor {
                     if (stepperOutcome.next_peer_pks.length > 0) {
                         this.strategy.sendParticleFurther(newParticle);
                     }
+
+                    if (stepperOutcome.ret_code == 0) {
```

Comment:
We can probably write something like this `stepperOutcome.ret_code == 0 && resolve`